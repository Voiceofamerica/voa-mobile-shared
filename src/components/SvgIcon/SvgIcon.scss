
.icon {
  display: inline-block;
  width: 1em;
  height: 1em;
  background: currentColor;
  mask-size: contain;
  -webkit-mask-size: contain;
  mask-repeat: no-repeat;
  -webkit-mask-repeat: no-repeat;
  mask-position: center;
  -webkit-mask-position: center;
}
