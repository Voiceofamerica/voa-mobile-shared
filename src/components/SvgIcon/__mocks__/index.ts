
import * as React from 'react'

export default (props) => (
  React.createElement('svg-icon', { props }, props.children)
)
