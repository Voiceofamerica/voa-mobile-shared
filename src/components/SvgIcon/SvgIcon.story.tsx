
import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { text, number, withKnobs } from '@storybook/addon-knobs'
import ArticleBlurb from '../../types/ArticleBlurb'

import SvgIcon from './SvgIcon'

storiesOf('SvgIcon', module)
  .addDecorator(withKnobs)
  .add('display', () => (
    <SvgIcon src={require('./arrow-right.svg')} style={{ fontSize: number('Size', 100, { range: true, min: 10, max: 500, step: 10 }), color: text('Color', '#53A68E') }} />
  ))
