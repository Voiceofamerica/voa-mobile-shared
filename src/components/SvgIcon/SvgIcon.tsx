
import * as React from 'react'

import { icon } from './SvgIcon.scss'

export interface Props {
  src: string
  className?: string
  style?: React.CSSProperties
}

export default (props: Props) => {
  const { src, className = '', style = {} } = props

  const webkitMaskImage = `url(${src})`

  return (
    <div className={`${icon} ${className}`} style={{ ...style, webkitMaskImage }} />
  )
}
