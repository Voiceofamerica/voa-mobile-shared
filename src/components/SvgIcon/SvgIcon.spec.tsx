
import * as React from 'react'
import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'

import SvgIcon from './SvgIcon'

describe('<Spinner />', () => {
  describe('shapshots', () => {
    it('should render', () => {
      const element = create((
        <SvgIcon src='https://some.url/something.svg' />
      ))

      expect(element).toMatchSnapshot()
    })
    it('should render with the given classname and style', () => {
      const element = create((
        <SvgIcon src='https://some.url/something.svg' className='myClass' style={{ color: 'blue' }} />
      ))

      expect(element).toMatchSnapshot()
    })
  })
})
