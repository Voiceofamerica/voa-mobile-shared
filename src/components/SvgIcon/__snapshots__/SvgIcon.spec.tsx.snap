// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> shapshots should render 1`] = `
<div
  className="icon "
  style={
    Object {
      "webkitMaskImage": "url(https://some.url/something.svg)",
    }
  }
/>
`;

exports[`<Spinner /> shapshots should render with the given classname and style 1`] = `
<div
  className="icon myClass"
  style={
    Object {
      "color": "blue",
      "webkitMaskImage": "url(https://some.url/something.svg)",
    }
  }
/>
`;
